packages/eyes/src/eyes-test/eyes.ts

```diff
@@ -0,0 +1,170 @@
+  const isNew = status === 'new';
+  const isError = status === 'error';
+
+  const formattedUrl = url && isModified ? `${chalk.cyan('URL')}: ${chalk.underline(url)}` : '';
+  const statusMessage = isModified ? chalk.red('MODIFIED') :
+               isNew ? chalk.green('NEW') :
+               isError ? chalk.bgRedBright('ERROR') :
+               chalk.green('UNMODIFIED');
+  consoleLog(`${statusMessage} ${name} ${formattedUrl}`);
+
+  if (isError) {
+    consoleLog(error);
+  }
+}
+
+// We cannot allow any of the result promises to reject, because we're going to
+// await on a bunch of them in parallel using Promise.all, and a single rejected
+// promise would reject the entire batch.
+function getTestResult(testName: string, testResult: Promise<any>): Promise<IResult> {
+  return (
+    testResult
+    .catch((err: any) => err)
+    .then((res: any) => res instanceof TestFailedError ? res.getTestResults() : res)
+    .then((res: any) => Array.isArray(res) && res[0] instanceof TestResults ? res[0] : res)
+    .then((res: any): IResult => {
+      if (res instanceof TestResults) {
+        const url = res.getUrl();
+        if (res.getIsDifferent()) {
+          return {name: testName, status: 'modified', url};
+        }
+        if (res.getIsNew()) {
+          return {name: testName, status: 'new', url};
+        }
+        if (res.isPassed()) {
+          return {name: testName, status: 'unmodified', url};
+        }
+      }
+
+      return {name: testName, status: 'error', error: res};
+    })
+  );
+}
+
+async function runTest(gridClient: any, gridClientConfig: any, testName: string, html: string, resources: {[url: string]: IResource}) {
+  const dom = new JSDOM(html).window.document;
+
+  const {checkWindow, close} = await gridClient.openEyes({
+    ...gridClientConfig,
+    testName,
+  });
+
+  await checkWindow({
+    url: 'http://localhost/html/index.html',
+    cdt: domNodesToCdt(dom),
+    sizeMode: 'viewport',
+    resourceContents: resources
+  });
+
+  return close();
+}
+
+export async function runEyes(projectPath: string, directory: string) {
+  const cssFilenames  = glob.sync('*.css', {cwd: directory});
+  const htmlFilenames = glob.sync('*.snapshot.html', {cwd: directory});
+
+  const config = getGridClientConfig(projectPath);
+  const resources = getStaticResources(cssFilenames, directory);
+  const gridClient = makeVisualGridClient(initConfig());
+
+  const resultPromises = [];
+  consoleLog('Sending snapshots to Applitools...');
+
+  for (const htmlFilename of htmlFilenames) {
+    const html = fs.readFileSync(path.join(directory, htmlFilename), 'utf-8');
+    const testName = formatName(htmlFilename);
+
+    const result = getTestResult(testName, runTest(
+      gridClient,
+      config,
+      testName,
+      html,
+      resources
+    ));
+
+    result.then((res: IResult) => {
+      if (res.status === 'error' || res.status === 'modified') {
+        process.exitCode = 1;
+        logEyesResult(res);
```

Comment:
Now I'm thinking that we don't really need to pass testName to `getTestResults` only to receive it back again, and instead we can pass it here: `logEyesResult(testName, res)`